events/syscall/utils_linux.go

```diff
@@ -91,3 +93,24 @@ func runAsUser(h events.Helper, username string, cmdName string, cmdArgs ...stri
 	}
 	return cmd.Run()
 }
+
+// Creates a temp directory and .ssh directory inside it.
+func createSshDirectoryUnderHome() (string, error) {
+	// Creates temporary data for testing.
+	var (
+		tempDirectoryName string
+		err               error
+	)
+	// Loop until a unique temporary directory is successfully created
+	if tempDirectoryName, err = os.MkdirTemp("/home", "falco-event-generator-"); err != nil {
+		return "", err
+	}
+
+	// Create the SSH directory
+	sshDir := filepath.Join(tempDirectoryName, ".ssh")
+	if err := os.Mkdir(sshDir, 0755); err != nil {
+		return "", err
+	}
+
+	return tempDirectoryName, nil
```

Comment:
```suggestion
	return sshDir, nil
```

Reply:
I got https://github.com/falcosecurity/event-generator/pull/112#discussion_r1547453475, what if this function also returns a `cleanup` function?, like:
```go
return sshDir, func() { _ = os.RemoveAll(tempDirectoryName) }, nil
```

then the caller will just `defer cleanup()` after it captures the result, of course if cleanup != nil.
This is often used in go :)

Reply:
Yepp!! Awesome